File: Estruturais/Adapter/Python/Adapter.py
class UI:
    RESET = "\033[0m"
    BOLD = "\033[1m"
    PINK = "\033[38;5;213m"
    PURPLE = "\033[38;5;99m"
    WHITE = "\033[97m"
    GREEN = "\033[38;5;84m"
    YELLOW = "\033[38;5;228m"
    RED = "\033[38;5;203m"
    CYAN = "\033[38;5;117m"


class Pizza:

    def __init__(self, tamanho, borda, sabor):
        self.tamanho = tamanho
        self.borda = borda
        self.sabor = sabor

    def mostrar_pizza(self):

        print(
            UI.PURPLE +
            "\n╔════════════════════════════════════════════════════╗" +
            UI.RESET
        )

        print(
            UI.PURPLE +
            "║               🍕 PIZZA MONTADA 🍕                ║" +
            UI.RESET
        )

        print(
            UI.PURPLE +
            "╠══════════════╦════════════════╦════════════════════╣" +
            UI.RESET
        )

        print(
            UI.PURPLE + "║ " +
            UI.YELLOW + "Tamanho".ljust(12) +
            UI.PURPLE + " ║ " +
            UI.YELLOW + "Borda".ljust(14) +
            UI.PURPLE + " ║ " +
            UI.YELLOW + "Sabor".ljust(18) +
            UI.PURPLE + " ║" +
            UI.RESET
        )

        print(
            UI.PURPLE +
            "╠══════════════╬════════════════╬════════════════════╣" +
            UI.RESET
        )

        print(
            UI.PURPLE + "║ " +
            UI.WHITE + self.tamanho.ljust(12) +
            UI.PURPLE + " ║ " +
            UI.WHITE + self.borda.ljust(14) +
            UI.PURPLE + " ║ " +
            UI.WHITE + self.sabor.ljust(18) +
            UI.PURPLE + " ║" +
            UI.RESET
        )

        print(
            UI.PURPLE +
            "╚══════════════╩════════════════╩════════════════════╝" +
            UI.RESET
        )


class SistemaAntigoPizza:

    def enviar_pedido(self, tamanho, borda, sabor):

        print(
            UI.GREEN +
            "\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━" +
            UI.RESET
        )

        print(
            UI.GREEN +
            UI.BOLD +
            "✅ PEDIDO ENVIADO PARA O SISTEMA ANTIGO" +
            UI.RESET
        )

        print(
            UI.WHITE +
            "\n📦 Dados convertidos:" +
            UI.RESET
        )

        print(
            UI.CYAN +
            f"➜ {tamanho} | {borda} | {sabor}" +
            UI.RESET
        )

        print(
            UI.GREEN +
            "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n" +
            UI.RESET
        )


class PizzaAdapter:

    def __init__(self, sistema_antigo):
        self.sistema_antigo = sistema_antigo

    def criar_pizza(self, pizza):

        self.sistema_antigo.enviar_pedido(
            pizza.tamanho,
            pizza.borda,
            pizza.sabor
        )


def linha():

    print(
        UI.PURPLE +
        "══════════════════════════════════════════════════════" +
        UI.RESET
    )


def erro():

    print(
        UI.RED +
        "\n❌ Opção inválida! Escolha uma opção do menu.\n" +
        UI.RESET
    )


def escolher_tamanho():

    while True:

        print(
            UI.PINK +
            UI.BOLD +
            "\n🍕 ESCOLHA O TAMANHO DA PIZZA\n" +
            UI.RESET
        )

        print(UI.WHITE + "[1]" + UI.RESET + " Brotinho")
        print(UI.WHITE + "[2]" + UI.RESET + " Tradicional")
        print(UI.WHITE + "[3]" + UI.RESET + " Grande")

        opcao = input(
            UI.CYAN + "\n➜ Digite sua opção: " + UI.RESET
        )

        if opcao == "1":
            return "Brotinho"

        elif opcao == "2":
            return "Tradicional"

        elif opcao == "3":
            return "Grande"

        else:
            erro()


def escolher_borda():

    while True:

        linha()

        print(
            UI.PINK +
            UI.BOLD +
            "\n🧀 ESCOLHA A BORDA\n" +
            UI.RESET
        )

        print(UI.WHITE + "[1]" + UI.RESET + " Tradicional")
        print(UI.WHITE + "[2]" + UI.RESET + " Catupiry")
        print(UI.WHITE + "[3]" + UI.RESET + " Sem borda")

        opcao = input(
            UI.CYAN + "\n➜ Digite sua opção: " + UI.RESET
        )

        if opcao == "1":
            return "Tradicional"

        elif opcao == "2":
            return "Catupiry"

        elif opcao == "3":
            return "Sem borda"

        else:
            erro()


def escolher_sabor():

    while True:

        linha()

        print(
            UI.PINK +
            UI.BOLD +
            "\n🍕 ESCOLHA O SABOR\n" +
            UI.RESET
        )

        print(UI.WHITE + "[1]" + UI.RESET + " Calabresa")
        print(UI.WHITE + "[2]" + UI.RESET + " Frango")
        print(UI.WHITE + "[3]" + UI.RESET + " Portuguesa")

        opcao = input(
            UI.CYAN + "\n➜ Digite sua opção: " + UI.RESET
        )

        if opcao == "1":
            return "Calabresa"

        elif opcao == "2":
            return "Frango"

        elif opcao == "3":
            return "Portuguesa"

        else:
            erro()


print(
    UI.PURPLE +
    UI.BOLD +
    "\n══════════════════════════════════════════════════════" +
    "\n              🍕 PIZZARIA ADAPTER 🍕               " +
    "\n══════════════════════════════════════════════════════" +
    UI.RESET
)

tamanho = escolher_tamanho()
borda = escolher_borda()
sabor = escolher_sabor()

pizza = Pizza(
    tamanho,
    borda,
    sabor
)

pizza.mostrar_pizza()

sistema_antigo = SistemaAntigoPizza()

adapter = PizzaAdapter(
    sistema_antigo
)

adapter.criar_pizza(pizza)

File: Estruturais/Adapter/JavaScript/Adapter.js
const readline = require("readline");

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const ui = {
    reset: "\x1b[0m",
    bold: "\x1b[1m",
    pink: "\x1b[38;5;213m",
    purple: "\x1b[38;5;99m",
    white: "\x1b[97m",
    green: "\x1b[38;5;84m",
    yellow: "\x1b[38;5;228m",
    red: "\x1b[38;5;203m",
    cyan: "\x1b[38;5;117m"
};

class Pizza {
    constructor(tamanho, borda, sabor) {
        this.tamanho = tamanho;
        this.borda = borda;
        this.sabor = sabor;
    }

    mostrarPizza() {

        console.log(
            ui.purple +
            "\n╔════════════════════════════════════════════════════╗" +
            ui.reset
        );

        console.log(
            ui.purple +
            "║               🍕 PIZZA MONTADA 🍕                ║" +
            ui.reset
        );

        console.log(
            ui.purple +
            "╠══════════════╦════════════════╦════════════════════╣" +
            ui.reset
        );

        console.log(
            ui.purple + "║ " +
            ui.yellow + "Tamanho".padEnd(12) +
            ui.purple + " ║ " +
            ui.yellow + "Borda".padEnd(14) +
            ui.purple + " ║ " +
            ui.yellow + "Sabor".padEnd(18) +
            ui.purple + " ║" +
            ui.reset
        );

        console.log(
            ui.purple +
            "╠══════════════╬════════════════╬════════════════════╣" +
            ui.reset
        );

        console.log(
            ui.purple + "║ " +
            ui.white + this.tamanho.padEnd(12) +
            ui.purple + " ║ " +
            ui.white + this.borda.padEnd(14) +
            ui.purple + " ║ " +
            ui.white + this.sabor.padEnd(18) +
            ui.purple + " ║" +
            ui.reset
        );

        console.log(
            ui.purple +
            "╚══════════════╩════════════════╩════════════════════╝" +
            ui.reset
        );
    }
}

class SistemaAntigoPizza {

    enviarPedido(tamanho, borda, sabor) {

        console.log(
            ui.green +
            "\n━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━" +
            ui.reset
        );

        console.log(
            ui.green +
            ui.bold +
            "✅ PEDIDO ENVIADO PARA O SISTEMA ANTIGO" +
            ui.reset
        );

        console.log(
            ui.white +
            "\n📦 Dados convertidos:" +
            ui.reset
        );

        console.log(
            ui.cyan +
            "➜ " + tamanho +
            " | " + borda +
            " | " + sabor +
            ui.reset
        );

        console.log(
            ui.green +
            "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\n" +
            ui.reset
        );
    }
}

class PizzaAdapter {

    constructor(sistemaAntigo) {
        this.sistemaAntigo = sistemaAntigo;
    }

    criarPizza(pizza) {
        this.sistemaAntigo.enviarPedido(
            pizza.tamanho,
            pizza.borda,
            pizza.sabor
        );
    }
}

function linha() {

    console.log(
        ui.purple +
        "══════════════════════════════════════════════════════" +
        ui.reset
    );
}

function erro() {

    console.log(
        ui.red +
        "\n❌ Opção inválida! Escolha uma opção do menu.\n" +
        ui.reset
    );
}

function escolherTamanho() {

    console.log(
        ui.pink +
        ui.bold +
        "\n🍕 ESCOLHA O TAMANHO DA PIZZA\n" +
        ui.reset
    );

    console.log(ui.white + "[1]" + ui.reset + " Brotinho");
    console.log(ui.white + "[2]" + ui.reset + " Tradicional");
    console.log(ui.white + "[3]" + ui.reset + " Grande");

    rl.question(
        ui.cyan + "\n➜ Digite sua opção: " + ui.reset,
        (opTamanho) => {

            let tamanho = "";

            if (opTamanho === "1") {
                tamanho = "Brotinho";
            }

            else if (opTamanho === "2") {
                tamanho = "Tradicional";
            }

            else if (opTamanho === "3") {
                tamanho = "Grande";
            }

            else {
                erro();
                return escolherTamanho();
            }

            escolherBorda(tamanho);
        }
    );
}

function escolherBorda(tamanho) {

    linha();

    console.log(
        ui.pink +
        ui.bold +
        "\n🧀 ESCOLHA A BORDA\n" +
        ui.reset
    );

    console.log(ui.white + "[1]" + ui.reset + " Tradicional");
    console.log(ui.white + "[2]" + ui.reset + " Catupiry");
    console.log(ui.white + "[3]" + ui.reset + " Sem borda");

    rl.question(
        ui.cyan + "\n➜ Digite sua opção: " + ui.reset,
        (opBorda) => {

            let borda = "";

            if (opBorda === "1") {
                borda = "Tradicional";
            }

            else if (opBorda === "2") {
                borda = "Catupiry";
            }

            else if (opBorda === "3") {
                borda = "Sem borda";
            }

            else {
                erro();
                return escolherBorda(tamanho);
            }

            escolherSabor(tamanho, borda);
        }
    );
}

function escolherSabor(tamanho, borda) {

    linha();

    console.log(
        ui.pink +
        ui.bold +
        "\n🍕 ESCOLHA O SABOR\n" +
        ui.reset
    );

    console.log(ui.white + "[1]" + ui.reset + " Calabresa");
    console.log(ui.white + "[2]" + ui.reset + " Frango");
    console.log(ui.white + "[3]" + ui.reset + " Portuguesa");

    rl.question(
        ui.cyan + "\n➜ Digite sua opção: " + ui.reset,
        (opSabor) => {

            let sabor = "";

            if (opSabor === "1") {
                sabor = "Calabresa";
            }

            else if (opSabor === "2") {
                sabor = "Frango";
            }

            else if (opSabor === "3") {
                sabor = "Portuguesa";
            }

            else {
                erro();
                return escolherSabor(tamanho, borda);
            }

            const pizza = new Pizza(
                tamanho,
                borda,
                sabor
            );

            pizza.mostrarPizza();

            const sistemaAntigo = new SistemaAntigoPizza();

            const adapter = new PizzaAdapter(
                sistemaAntigo
            );

            adapter.criarPizza(pizza);

            rl.close();
        }
    );
}

console.clear();

console.log(
    ui.purple +
    ui.bold +
    "\n══════════════════════════════════════════════════════" +
    "\n              🍕 PIZZARIA ADAPTER 🍕                   " +
    "\n══════════════════════════════════════════════════════" +
    ui.reset
);

escolherTamanho();